akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/adapter/AdapterSpec.scala

```diff
@@ -157,6 +157,36 @@ class AdapterSpec extends AkkaSpec {
 
       typed1 should be theSameInstanceAs typed2
     }
+
+    "not crash if guardian is stopped" in {
+      for { _ ← 0 to 10 } {
+        var system: akka.actor.typed.ActorSystem[NotUsed] = null
+        try {
+          system = ActorSystem.create(Behavior.stopped[NotUsed], "AdapterSpec-stopping-guardian")
+
+        } finally {
+          if (system != null) Await.result(system.terminate(), remainingOrDefault)
+        }
+      }
+    }
+
+    "not crash if guardian is stopped very quickly" in {
+      for { _ ← 0 to 10 } {
+        var system: akka.actor.typed.ActorSystem[Done] = null
+        try {
+          system = ActorSystem.create(Actor.immutable[Done] { (ctx, msg) ⇒
+            ctx.self ! Done
+            msg match {
+              case Done ⇒ Actor.stopped
+            }
+
+          }, "AdapterSpec-stopping-guardian-2")
+
+        } finally {
+          if (system != null) Await.result(system.terminate(), remainingOrDefault)
+        }
+      }
+    }
   }
```

Comment:
Perhaps add a unit test for `delayedGuardianStart` to verify the stashing. I think it can be tested without being the actual guardian.

Reply:
Ah, clever, didn't think of that. Will do.